src/cairoUtilFuncGen/abi/base.ts

```diff
@@ -45,3 +47,22 @@ export abstract class AbiBase extends StringIndexedFuncGenWithAuxiliar {
     throw new Error('Method not implemented.');
   }
 }
+
+/**
+ * Returns a static array type string without the element
+ * information
+ * e.g.
+ *    uint8[20] -> uint8[]
+ *    uint[][8] -> uint[][]
+ *    uint[10][15] -> uint[10][]
+ *  @param type a static ArrayType
+ *  @returns static array without length information
+ */
+export function removeSizeInfo(type: ArrayType): string {
+  assert(type.size !== undefined, 'Expected an ArrayType with known type (a solc static array)');
+  const typeString = type.pp();
+  const reversedTypeString = typeString.split('').reverse().join('');
```

Comment:
`stringVal.replaceAll(/\[\d+\]/g, '[]')`
Why not something like this ^?

Reply:
I made this to have equal typestrings from static arrays, that receive the length as an input so we can use the same generated function for both

Reply:
The purpose of the function is to replace the length from static arrays.
So that generated functions for T[x] && T[y] could be the same and the length is used as a paremeter. This way we avoid the generation of similar functions with just a single value change (x -> y)

I do all the reversing and stuff so that I only take the last one:
T[z][x] -> T[z][].

I don't want to remove them all because I still need to keep the static length for nested types. So that the generated function of T[z][] can call to the generated function of T[z] (which will be T[]) but with size `z`.

Reply:
The purpose of the function is to replace the length from static arrays.
So that generated functions for T[x] && T[y] could be the same and the length is used as a paremeter. This way we avoid the generation of similar functions with just a single value change (x -> y)

I do all the reversing and stuff so that I only take the last one:
T[z][x] -> T[z][].

I don't want to remove them all because I still need to keep the static length for nested types. So that the generated function of T[z][] can call to the generated function of T[z] (which will be T[]) but with size `z`.

Reply:
I see it makes sense